src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/TseViewApplicationHelper.java

```diff
@@ -0,0 +1,283 @@
+                        String documentLink = matcher.replaceFirst("");
+                        String documentName = doc.getValue().getUploadedDocument().getDocumentFilename();
+                        return String.format("<a href=\"/documents/%s\" target=\"_blank\">%s</a>", documentLink,
+                                documentName) + STRING_BR;
+                    }).collect(Collectors.joining(""));
+        }
+        return null;
+    }
+
+    private static String createAdminResponse(TseRespondTypeItem response, AtomicInteger count) {
+        String doc = "N/A";
+        if (response.getValue().getAddDocument() != null) {
+            Pattern pattern = Pattern.compile("^.+?/documents/");
+            Matcher matcher = pattern.matcher(response.getValue().getAddDocument().getDocumentBinaryUrl());
+            String documentLink = matcher.replaceFirst("");
+            String documentName = response.getValue().getAddDocument().getDocumentFilename();
+            doc = String.format("<a href=\"/documents/%s\" target=\"_blank\">%s</a>", documentLink,
+                    documentName);
+        }
+        return String.format(
+                VIEW_APPLICATION_ADMIN_REPLY_MARKUP,
+                count.get(),
+                formatAminResponseTitle(response.getValue()),
+                response.getValue().getDate(),
+                defaultString(response.getValue().getIsCmoOrRequest()),
+                formatResponseDue(response.getValue()),
+                formatPartyOrPartiesToRespond(response.getValue()),
+                formatAdminResponseAdditionalInfo(response.getValue()),
+                doc,
+                formatAdminReplyMadeBy(response.getValue()),
+                formatMadeByFullName(response.getValue()),
+                defaultString(response.getValue().getSelectPartyNotify()));
+    }
+
+    private static String createRespondentOrClaimantResponse(TseRespondTypeItem response,
+                                                             AtomicInteger count, String applicant) {
+        String doc = "N/A";
+
+        String links = getDocumentUrls(response);
+        if (links != null) {
+            doc = links;
+        }
+        return String.format(
+                RESPONDENT_REPLY_MARKUP_FOR_REPLY,
+                count.get(),
+                response.getValue().getFrom(),
+                response.getValue().getDate(),
+                applicant.toLowerCase(Locale.ENGLISH),
+                isNullOrEmpty(response.getValue().getResponse()) ? "N/A"
+                        : defaultString(response.getValue().getResponse()),
+                doc,
+                formatRule92(response.getValue().getCopyToOtherParty(),
+                        isNullOrEmpty(response.getValue().getCopyNoGiveDetails())
+                                ? "N/A" : response.getValue().getCopyNoGiveDetails()));
+    }
```

Comment:
This is kinda hard to read, could maybe split up the ternarys into their own variables and reference them here. Also theres many calls to `response.getValue()` this could be extracted into its own variable

Reply:
updated